services/categories.js

```diff
@@ -21,6 +21,24 @@ const restrictedFields = ['clickable'];
 const defaultFields    = ['_id', 'code', 'action', 'translation'];
 const queryBuilder     = new QueryBuilder(Categories, restrictedFields, defaultFields);
 
+const checkSlugExist = async (doc) => {
+    const arg = {$or: []};
+    if (doc.translation) {
+        for (const lang of Object.entries(doc.translation)) {
```

Comment:
instead of :
```js
for (const lang of Object.entries(doc.translation)) {
  if (doc.translation[lang[0]]) {
    arg.$or.push({[`translation.${lang[0]}.slug`]: doc.translation[lang[0]].slug});
  }
}
```
you could write it like that :
```js
// descructuring resulting array from Object.entries
for (const [lang] of Object.entries(doc.translation)) {
  if (doc.translation[lang]) {
    arg.$or.push({[`translation.${lang}.slug`]: doc.translation[lang].slug});
  }
}
```